crates/bevy_pbr/src/meshlet/asset.rs

```diff
@@ -68,84 +71,147 @@ pub struct MeshletBoundingSpheres {
 }
 
 /// A spherical bounding volume used for a [`Meshlet`].
-#[derive(Serialize, Deserialize, Copy, Clone, Pod, Zeroable)]
+#[derive(Copy, Clone, Pod, Zeroable)]
 #[repr(C)]
 pub struct MeshletBoundingSphere {
     pub center: Vec3,
     pub radius: f32,
 }
 
 /// An [`AssetLoader`] and [`AssetSaver`] for `.meshlet_mesh` [`MeshletMesh`] assets.
-pub struct MeshletMeshSaverLoad;
+pub struct MeshletMeshSaverLoader;
 
-impl AssetLoader for MeshletMeshSaverLoad {
+impl AssetSaver for MeshletMeshSaverLoader {
     type Asset = MeshletMesh;
     type Settings = ();
+    type OutputLoader = Self;
     type Error = MeshletMeshSaveOrLoadError;
 
-    async fn load<'a>(
+    async fn save<'a>(
         &'a self,
-        reader: &'a mut dyn Reader,
-        _settings: &'a Self::Settings,
-        _load_context: &'a mut LoadContext<'_>,
-    ) -> Result<Self::Asset, Self::Error> {
-        let version = read_u64(reader).await?;
-        if version != MESHLET_MESH_ASSET_VERSION {
-            return Err(MeshletMeshSaveOrLoadError::WrongVersion { found: version });
-        }
-
-        let mut bytes = Vec::new();
-        reader.read_to_end(&mut bytes).await?;
-        let asset = bincode::deserialize_from(FrameDecoder::new(Cursor::new(bytes)))?;
+        writer: &'a mut Writer,
+        asset: SavedAsset<'a, MeshletMesh>,
+        _settings: &'a (),
+    ) -> Result<(), MeshletMeshSaveOrLoadError> {
+        // Write asset version
```

Comment:
I'd suggest picking a randomly generated 8-byte "magic number" to write as the first thing in the file. That'll make it possible to identify whether a random file matches the format, since Bevy will be the only program that ever writes files with those initial bytes (with extremely high probability).

Reply:
Done